File: ISteamNetworking001.cs

using System;
using System.Runtime.InteropServices;

namespace SteamStructs {
    [StructLayout(LayoutKind.Sequential, Pack = 1)]
    internal struct ISteamNetworking001
    {
        public IntPtr CreateListenSocket;
        public IntPtr CreateP2PConnectionSocket;
        public IntPtr CreateConnectionSocket;
        public IntPtr DestroySocket;
        public IntPtr DestroyListenSocket;
        public IntPtr SendDataOnSocket;
        public IntPtr IsDataAvailableOnSocket;
        public IntPtr RetrieveDataFromSocket;
        public IntPtr IsDataAvailable;
        public IntPtr RetrieveData;
        public IntPtr GetSocketInfo;
        public IntPtr GetListenSocketInfo;
    }
}

File: ISteamNetworking001.cc
class ISteamNetworking001 {
	virtual unknown_ret CreateListenSocket(int, unsigned int, unsigned short) = 0;
	virtual unknown_ret CreateP2PConnectionSocket(CSteamID, int, int) = 0;
	virtual unknown_ret CreateConnectionSocket(unsigned int, unsigned short, int) = 0;
	virtual unknown_ret DestroySocket(unsigned int, bool) = 0;
	virtual unknown_ret DestroyListenSocket(unsigned int, bool) = 0;
	virtual unknown_ret SendDataOnSocket(unsigned int, void*, unsigned int, bool) = 0;
	virtual unknown_ret IsDataAvailableOnSocket(unsigned int, unsigned int*) = 0;
	virtual unknown_ret RetrieveDataFromSocket(unsigned int, void*, unsigned int, unsigned int*) = 0;
	virtual unknown_ret IsDataAvailable(unsigned int, unsigned int*, unsigned int*) = 0;
	virtual unknown_ret RetrieveData(unsigned int, void*, unsigned int, unsigned int*, unsigned int*) = 0;
	virtual unknown_ret GetSocketInfo(unsigned int, CSteamID*, int*, unsigned int*, unsigned short*) = 0;
	virtual unknown_ret GetListenSocketInfo(unsigned int, unsigned int*, unsigned short*) = 0;
};
